Source/santad/EventProviders/SNTEndpointSecurityManager.mm

```diff
@@ -0,0 +1,304 @@
+        sleep(30);
+        exit(ret);
+      default:
+        LOGE(@"Unable to create es client: %d. Sleeping for a minute.", ret);
+        sleep(60);
+        continue;
+    }
+  }
+}
+
+- (void)messageHandler:(const es_message_t *)m API_AVAILABLE(macos(10.15)) {
+  // TODO(bur/rah): Currently this class only subscribes to exec events. Move this code
+  // somewhere exec specific if other types of events are added to this client.
+  santa_message_t sm;
+
+  audit_token_t audit_token = {};
+  void (^callback)(santa_message_t);
+
+  switch (m->event_type) {
+    case ES_EVENT_TYPE_AUTH_EXEC: {
+//      // TODO(bur/rah): Probably also want to do this for is_es_client.
+//      // TODO(bur/rah): Since these events are not evaluated by Santa's pipline they are
+//      //                missing bits of information such as SHA256 and REASON. Refactor the
+//      //                logging cache.
+//      if (m->event.exec.target->is_platform_binary) {
+//        LOGD(@"platform binary: %s", sm.path);
+//        [self postAction:ACTION_RESPOND_ALLOW forMessage:sm];
+//        return;
+//      }
+      sm.es_message = (void *)es_copy_message(m);
+      sm.action = ACTION_REQUEST_BINARY;
+      sm.vnode_id.fsid = m->event.exec.target->executable->stat.st_dev;
+      sm.vnode_id.fileid = m->event.exec.target->executable->stat.st_ino;
+      callback = self.decisionCallback;
+      audit_token = m->event.exec.target->audit_token;
+      sm.ppid = m->event.exec.target->original_ppid;
+
+      size_t l = m->event.exec.target->executable->path.length;
+      if (l + 1 > MAXPATHLEN ||  m->event.exec.target->executable->path_truncated) {
+        // TODO(bur/rah): Get path from fsid.
+        LOGE(@"Path is truncated!");
+        es_respond_auth_result(self.client, m, ES_AUTH_RESULT_ALLOW, true);
+        return;
+      }
+      strncpy(sm.path,m->event.exec.target->executable->path.data, l);
+      sm.path[l] = '\0';
+
+      break;
+    }
+    case ES_EVENT_TYPE_NOTIFY_EXEC: {
+      sm.action = ACTION_NOTIFY_EXEC;
+      sm.vnode_id.fsid = m->event.exec.target->executable->stat.st_dev;
+      sm.vnode_id.fileid = m->event.exec.target->executable->stat.st_ino;
+
+      // TODO(rah): Profile this, it might need to be improved.
+      uint32_t argCount = es_exec_arg_count(&(m->event.exec));
+      NSMutableArray *args = [NSMutableArray arrayWithCapacity:argCount];
+      for (int i = 0; i < argCount; ++i) {
+        es_string_token_t arg = es_exec_arg(&(m->event.exec), i);
+        [args addObject:[[NSString alloc] initWithBytes:arg.data
+                                                 length:arg.length
+                                               encoding:NSUTF8StringEncoding]];
+      }
+      sm.args_array = (void *)CFBridgingRetain(args);
+
+      callback = self.logCallback;
+      audit_token = m->event.exec.target->audit_token;
+      sm.ppid = m->event.exec.target->original_ppid;
+
+      size_t l = m->event.exec.target->executable->path.length;
```

Comment:
What if `path_truncated` is true? Same throughout.

Reply:
For logging I don't think it particularly matters but we can re-visit when we refactor this.